### PR TITLE
Revise README

### PR DESCRIPTION
**Pull request by traversaro**
A README revamp was long overdue, so I did it.

Main points:
* Add an introduction section, briefly explaining what iDynTree **is** and what it **is not**. 
* Simplify installation, focusing on how to install iDynTree via **conda** and via **robotology-superbuild**. The installation from source is still documented and all the related quirks are still documented, but in a separate `build-from-source.md` doc that is meant to be consumed by experienced users. 
* Removed details on how to install qt, YARP and ICUB as they are not anymore dependency of iDynTree
* Add documentation of tools such as `ìdyntree-model-info` and `idyntree-model-view`. 

I hope the README is more readable now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,178 +17,45 @@ iDynTree is written in C++ language, but thanks to [SWIG](http://www.swig.org/)
...
-If you need to install also `YARP` and `ICUB`, it is recommended that you install iDynTree via the [`robotology-superbuild`](https://github.com/robotology/robotology-superbuild). If instead you are not interested in the `YARP` and `ICUB` integration, you can easily install the rest of the dependencies using
-standard package managers.
-
-##### Windows
-
-On Windows we recommend to use  [`vcpkg`](https://github.com/Microsoft/vcpkg) C++ package manager to install iDynTree dependencies.
-
-###### vcpkg
-If you use [`vcpkg`](https://github.com/Microsoft/vcpkg), you can install all the required and optional dependencies of iDynTree using the following command:
+You can easily the library with [`conda`](https://github.com/conda-forge/osqp-eigen-feedstock) using the following command
```
Comment (jcarpent) [inline README.md]: *install

Comment (traversaro) [reply README.md]: Thanks! Fixed in https://github.com/robotology/idyntree/pull/932/commits/ad39853abc47640540461e3b8707cc4a027f7278 .

**Comment by traversaro** [comment]
@VenusPasandi @fabiodinatale @CarlottaSartore  One of the goal of this cleanup is to make the README more clear as I wanted also to work on some tools that could be useful when working with URDF models, so feel free to read the new README and tell me if it is clear enough!

**Comment by traversaro** [comment]
The "Usage in Python" section can be improved, but to be honest I am not super sure how it works, and if we are going to enable both SWIG and pybind11 backends, I would prefer to write it after we do that.

**Review by GiulioRomualdi — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,194 +1,82 @@
...
+iDynTree is a library of robots dynamics algorithms for control, estimation and simulation. It is specifically designed for free-floating robots, but it is possible to use it also  with fixed-base robots. 
 
-iDynTree is specifically designed for free-floating robots, but it is possible to use it also  with fixed-base robots.
+The **major characteristic features** of iDynTree are:
+* It is written in **C++**, with **Python** and **MATLAB** bindings.
+* It uses an  **undirected graph data structure** (`iDynTree::Model`) that is used to represent robots, to easily change the **base link** that you are using for your kinematics and dynamics computations without the need to reload your model or change your joint or link serializations. This is done as iDynTree was developed for floating-base robots such as humanoids, in which the most convenient choice of **base link** can change.
+* It contains support for **reading and writing URDF files** from a `iDynTree::Model`, making it useful to write tools that modify robot models and saves them back to file. This is done as iDynTree was developed to developed **tools for identification of kinematics and dynamics parameters**.
```
Comment (VenusPasandi) [inline README.md:9]: a typo: developed to developed

**Review by VenusPasandi — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (traversaro) [reply README.md]: Thanks, fixed in https://github.com/robotology/idyntree/pull/932/commits/830ad70afffd8df1b749cc04c2b78e9f01ef71a8 .

**Comment by traversaro** [comment]
I merged the PR, if you have further comment feel free to provide them enve if the PR was merged.

**Comment by traversaro** [comment]
I merged the PR, if you have further comment feel free to provide them even if the PR was merged.

File: README.md [hunk 3]
```diff
@@ -1,194 +1,82 @@
...
+https://pure.tue.nl/ws/portalfiles/portal/139293126/A_Multibody_Dynamics_Notation_Revision_2_.pdf).
+* It contains an implementation of the algorithm used in the iCub humanoid robot to **estimate the joint torques without the need of collocated joint torque sensors**, exploting the specific **undirected graph data structure** . This is done as this was one of the originally goal for the implementation of iDynTree. See the class [`iDynTree::ExtWrenchesAndJointTorquesEstimator`](https://github.com/robotology/idyntree/blob/master/src/estimation/include/iDynTree/Estimation/ExtWrenchesAndJointTorquesEstimator.h) and [Chapter 6 of "Modelling, Estimation and Identification of Humanoid Robots Dynamics"](https://traversaro.github.io/traversaro-phd-thesis/traversaro-phd-thesis.pdf).
+
+To avoid confusion, it is also useful to clarify what **iDynTree is not**:
+* It is not the **fastest C++ library** for kinematics and dynamics multibody computations for robotics. It is not slow, but if have an application in which you need the absolute fastest library, check out [Pinocchio](https://github.com/stack-of-tasks/pinocchio).
```
Comment (fabiodinatale) [inline README.md:15]: Typo mistakes: `...but if have an application...` **you** missing.

File: README.md [hunk 4]
```diff
@@ -1,194 +1,82 @@
...
- ./vcpkg install --triplet x64-windows assimp eigen3 qt5 libxml2 irrlicht
-~~~
-The default way to use the libraries provided by vcpkg in CMake is to use the [vcpkg CMake toolchain](https://github.com/Microsoft/vcpkg/blob/master/docs/users/integration.md#cmake-toolchain-file-recommended-for-open-source-cmake-projects).
-
-If you want also to install the `ipopt` library, we recommend to use the port `ipopt-binary` available at https://github.com/robotology/robotology-vcpkg-ports .
+### conda (recommended) 
 
-##### macOS
-You can install most of the required and optional dependencies of iDynTree using [homebrew](https://brew.sh/) with the following command:
+You can easily install the C++ and Python library with via [`conda-forge`](https://conda-forge.org) using the following command
```
Comment (fabiodinatale) [inline README.md:33]: Typo mistake: `with via`, I think there is a repetition.

**Pull request merged**